classes/core/PKPAppKey.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/**
+ * @file classes/core/PKPAppKey.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPAppKey
+ *
+ * @ingroup core
+ *
+ * @brief Class to manage app key related behaviours
+ */
+
+namespace PKP\core;
+
+use Exception;
+use PKP\config\Config;
+use PKP\config\ConfigParser;
+use Illuminate\Support\Str;
+use Illuminate\Encryption\Encrypter;
+
+class PKPAppKey
+{
+    /**
+     * The supported cipher algorithms and their properties.
+     *
+     * @var array
+     */
+    private static array $supportedCiphers = [
+        'aes-128-cbc' => ['size' => 16, 'aead' => false],
+        'aes-256-cbc' => ['size' => 32, 'aead' => false],
+        'aes-128-gcm' => ['size' => 16, 'aead' => true],
+        'aes-256-gcm' => ['size' => 32, 'aead' => true],
+    ];
+
+    /**
+     * The default cipher algorithms
+     *
+     * @var string
+     */
+    private static string $defaultCipher = 'aes-256-cbc';
+
+    /**
+     * Get the list of supported ciphers
+     */
+    public static function getSupportedCiphers(): array
```

Comment:
I don't think the getter function adds value; in general I'd like to decrease our use of getters/setters (and eventually [property hooks](https://wiki.php.net/rfc/property-hooks) will remove more justification for these). I don't have a strong opinion about static variables vs. constants, but we do have an existing pattern for `public const MY_CONST_NAME = abc` in classes, and the `UPPER_SNAKE_CASE` convention does make the invariability clear.